### PR TITLE
IQSS/11592-Fix Handle case insensitivity broken in 6.5

### LINKED ISSUE #11592: Handle.net PIDs - exception creating new dataset in Dataverse 6.5-6.6 after update from 6.4
Dear Dataverse team,
There is a problem creating a new dataset using Handles (Handle.net PID) since update to versions 6.5 and 6.6 - there is an exception about "UnmanagedHandleProvider" and dataset is not created, showing also an error page in GUI "Page Not Found - The page you are looking for was not found.":

```
[edu.harvard.iq.dataverse.DatasetPage] [tid: _ThreadID=102 _ThreadName=http-thread-pool::jk-connector(1)] [timeMillis: 1749748666538] [levelValue: 1000] [[
  CommandException, when attempting to update the dataset: PID Provider UnmanagedHandleProvider is not configured.
edu.harvard.iq.dataverse.engine.command.exception.IllegalCommandException: PID Provider UnmanagedHandleProvider is not configured.
        at edu.harvard.iq.dataverse.engine.command.impl.CreateNewDatasetCommand.handlePid(CreateNewDatasetCommand.java:100)
        at edu.harvard.iq.dataverse.engine.command.impl.AbstractCreateDatasetCommand.execute(AbstractCreateDatasetCommand.java:134)
        at edu.harvard.iq.dataverse.engine.command.impl.AbstractCreateDatasetCommand.execute(AbstractCreateDatasetCommand.java:26)
        at edu.harvard.iq.dataverse.EjbDataverseEngineInner.submit(EjbDataverseEngineInner.java:36)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:569)
```

I see in the history, that some changes for PIDs were done for 6.5 (before Dec 2024), but the most of them were done for other PID providers, not for Handles: https://github.com/IQSS/dataverse/commits/develop/src/main/java/edu/harvard/iq/dataverse/pidproviders?after=0177e5d9d4965dd97bfc361d8dace7872be49d78+34
Maybe some needed changes for Handles were skipped, because in version 6.4 the same Handle-config works as expected well.

**To whom does it occur (all users, curators, superusers)?
all users

**Which version of Dataverse are you using?**
The problem is in 6.5, 6.6. Versions 6.4 and below are correct.

**Any related open or closed issues to this bug report?**

There is a list of Dataverse instances which are using Handle.Net PIDs: #8881. The latest version of Dataverse there is 6.4. 

Communication in the user group: see [here](https://groups.google.com/g/dataverse-community/c/ARBMQTjivAU).

### PR DESCRIPTION
**Pull request by qqmyers**
**What this PR does / why we need it**: Prior changes to make DOIs fully case insensitive had a bug that also made Handles (which are case sensitive) fail PID recognition when the shoulder is specified as lower case. This PR corrects the bug to only use a toUpperCase() on the shoulder for case-insensitive PID types. It also updates the tests to use a non-empty Handle shoulder and verify that comparisons are case sensitive.

**Which issue(s) this PR closes**:

- Closes #11592

**Special notes for your reviewer**:

**Suggestions on how to test this**: Since we don't have a Handle setup, rely on the unit test, regression test. The submitter of the issue may be able to test the fix as well.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**: inc.

**Additional documentation**:

### CODE REVIEW

File: src/test/java/edu/harvard/iq/dataverse/pidproviders/PidUtilTest.java [hunk 1]
```diff
@@ -294,16 +294,22 @@ public void testDOIParsing() throws IOException {
...
         assertEquals("hdl1", pid2.getProviderId());
-        assertEquals("https://hdl.handle.net/" + pid2.getAuthority() + PidUtil.getPidProvider(pid2.getProviderId()).getSeparator() + pid2.getIdentifier(),pid2.asURL());
+        assertEquals("https://hdl.handle.net/" + pid2.getAuthority() + PidUtil.getPidProvider(pid2.getProviderId()).getSeparator() + pid2.getIdentifier(), pid2.asURL());
         assertEquals("20.500.1234", pid2.getAuthority());
         assertEquals(HandlePidProvider.HDL_PROTOCOL, pid2.getProtocol());
         GlobalId pid3 = PidUtil.parseAsGlobalID(pid2.asURL());
         assertEquals(pid1String, pid3.asString());
         assertEquals("hdl1", pid3.getProviderId());
+        
+        // Test case sensitivity - a handle with uppercase "TEST" should not match the hdl1 provider
```
Comment (stevenwinship) [inline src/test/java/edu/harvard/iq/dataverse/pidproviders/PidUtilTest.java:308]: This comment leads me to believe you will be comparing the uppercase "TEST" of pid4 to the lowercase "test" in pid1String showing they are not equal. 

Maybe add:
assertNotEquals(pid4.asString(), pid1String);

Comment (qqmyers) [reply src/test/java/edu/harvard/iq/dataverse/pidproviders/PidUtilTest.java:308]: ? - this test has nothing to do with pid1 other than all 4 tests use the hdl1 provider defined around line 110. Line 110 itself defines the hdl1 provider as having the lower case 'test' shoulder, so hdl4 which uses upper case won't match it. In this case, pid4String should still be recognized as a Handle, so the assertEquals on 311 should work (the string form of the pid equals the original string), but it won't be recognized by the hdl1 provider, only the UnmanagedHandlePidProvider (tested on line 312).

Comment (stevenwinship) [reply src/test/java/edu/harvard/iq/dataverse/pidproviders/PidUtilTest.java:308]: My point was there should be a negative test showing that "test" isn't equal to "TEST"

**Comment by ofahimIQSS** [comment]
Tests are passing - no issues found. Going to merge.
